### PR TITLE
Fix h5py issue in GHA test jobs

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** Fix h5py issue in GHA test jobs: `error: libhdf5.so: cannot open shared object file: No such file or directory`

**Ticket:** TBD

### CODE REVIEW

File: tools/mo/requirements_tf.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 -c ../constraints.txt
-tensorflow>=1.15.5,<2.17.0
+tensorflow>=1.15.5,<2.16.0
```
Comment (rkazants) [inline tools/mo/requirements_tf.txt:2]: I will probably revert this change. TF does not affect. And constraint on h5py looks to help resolving

File: tests/constraints.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 numpy>=1.16.6,<1.27
 attrs==23.2.0
 distro==1.9.0
-h5py>=3.1.0
+h5py>=3.1.0,<3.11.0
```
Comment (mryzhov) [inline tests/constraints.txt:4]: Please remove https://github.com/mryzhov/openvino/blob/c_unit_tests_fix/src/frontends/tensorflow/tests/requirements.txt#L5 too

File: src/bindings/python/constraints.txt [hunk 3]
```diff
@@ -18,7 +18,7 @@ patchelf<=0.17.2.1
 # Frontends
 docopt~=0.6.2
 paddlepaddle==2.5.2
-tensorflow>=1.15.5,<2.17.0
+tensorflow>=1.15.5,<2.16.0
```
Comment (rkazants) [inline src/bindings/python/constraints.txt:21]:
```suggestion
tensorflow>=1.15.5,<2.17.0
```

**Review by mryzhov — APPROVED** [review]

File: tools/mo/requirements_tf.txt [hunk 4]
```diff
@@ -1,5 +1,6 @@
 -c ../constraints.txt
-tensorflow>=1.15.5,<2.17.0
+h5py>=3.1.0,<3.11.0
+tensorflow>=1.15.5,<2.16.0
```
Comment (rkazants) [inline tools/mo/requirements_tf.txt:3]:
```suggestion
tensorflow>=1.15.5,<2.17.0
```

File: tools/mo/requirements_tf.txt [hunk 5]
```diff
@@ -1,4 +1,5 @@
 -c ../constraints.txt
+h5py>=3.1.0,<3.11.0
```
Comment (rkazants) [inline tools/mo/requirements_tf.txt:2]:
```suggestion
h5py
```

File: tools/mo/requirements_tf2.txt [hunk 6]
```diff
@@ -1,5 +1,6 @@
 -c ../constraints.txt
-tensorflow>=2.5,<2.17.0
+h5py
+tensorflow>=2.5,<2.16.0
```
Comment (rkazants) [inline tools/mo/requirements_tf2.txt:3]:
```suggestion
tensorflow>=2.5,<2.17.0
```

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
could you please port to 2023.3 LTS?

**Pull request merged**